README-National-Cloud.md

```diff
@@ -0,0 +1,193 @@
+
+#### Sign up
+
+- When running the app for the first time, you'd need to sign-in as an administrator first. Click the `Sign Up` link on the top bar.
+
+![Sign Up link](./ReadmeFiles/Sign-Up.JPG)
+
+- You will be presented with a form that simulates an onboarding process. Check the checkbox and  Click the `SignUp` button.
+
+![Sign Up form](./ReadmeFiles/Sign-Up-Admin.JPG)
+
+ - Now you are going through the [admin consent](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-convert-app-to-be-multi-tenant#admin-consent) flow. In this flow, the app gets provisioned for all the users in one organization. You'll be transferred to the Azure AD portal. Sign in as the administrator and you'd be presented with the following screen to consent on behalf of all users.
+
+![Admin Consent](./ReadmeFiles/Admin_consent.JPG)
+
+- Click `Accept` to provision a [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals#service-principal-object) of this app in the tenant.
+You will be transported back to the app, where your registration will be finalized.
+
+- If the app is not provisioned in your tenant by the tenant administrator using the steps laid out above, the sign-up process will result in the following error.
+
+![Need admin approval](./ReadmeFiles/NeedAdminApproval.JPG)
+
+- This step uses the `prompt=admin_consent' option provided in the [OAuth 2.0 authorization code grant](https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-protocols-oauth-code#request-an-authorization-code) to provide the administrator an option to consent for the entire tenant.
+
+- You can also use the [admin consent endpoint](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#using-the-admin-consent-endpoint) to provision the app in the chosen tenant.
+
+#### Sign in
+
+- Once you signed up, you can either click on the `Todo` tab or the `sign in` link to gain access to the application. 
+- If you are selecting **sign in the same session in which you signed up, you will automatically sign in with the same account you used for signing up.
+- If you are signing in during a new session, you will be presented with Azure AD's credentials prompt: sign in using an account compatible with the sign-up option you chose earlier (the exact same account if you used user consent, any user form the same tenant if you used admin consent).
+
+- If you try to sign-in before the tenant administrator has provisioned the app in the tenant using the `Sign up` link above, you will see the following error.
+
+![AADSTS700016, App not found](./ReadmeFiles/AADSTS700016.JPG)
+
+> To learn more about the code,  go to [About the code](./README.md#about-the-code) 
+
+## Community Help and Support
+
+Use [Stack Overflow](http://stackoverflow.com/questions/tagged/adal) to get support from the community.
```

Comment:
MSAL

Reply:
this one's v1 sample, we uses msal in v2 samples

---
In reply to: [254551172](https://github.com/Azure-Samples/active-directory-dotnet-webapp-multitenant-openidconnect/pull/37#discussion_r254551172) [](ancestors = 254551172)